src/SAML2/WebSSO/SP.hs

```diff
@@ -297,15 +297,17 @@ checkAssertions missuer (toList -> assertions) uref@(UserRef issuer _) = do
   when (null statements) $
     deny DeniedNoStatements  -- (not sure this is even possible?)
 
-  when (null . catMaybes $ (^? _AuthnStatement) <$> statements) $
+  unless (any isAuthnStatement statements) $
```

Comment:
this change looked necessary in some intermediate state of things, now it *could* be rolled back.  i kinda like the new version, though.

Reply:
I agree that this one is better;

For future reference the old version can be expressed a little clearer as 

```haskell
unless (has (traversed . _AuthnStatement) statements)
```